src/modules/tree/TreeView.tsx

```diff
@@ -0,0 +1,159 @@
+import { getNodeTree } from '@/utils/item';
+
+import './style.css';
+
+const MAX_NUM_ITEMS = 10;
+
+type Props = {
+  id: string;
+  header?: string;
+  items?: DiscriminatedItem[];
+  initialExpendedItems?: string[];
+  selectedId?: string;
+  onTreeItemSelect?: (value: string) => void;
+  isLoading?: boolean;
+  onlyShowContainerItems?: boolean;
+};
+
+// Props here is passed from TreeView react-accessible-treeview component
+interface NodeProps {
+  element: INode;
+  isBranch: boolean;
+  isExpanded: boolean;
+  isSelected: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  getNodeProps: () => any;
+  onSelect: (id: UUID) => void;
+}
+const RenderedNode = ({
+  element,
+  isBranch,
+  isExpanded,
+  getNodeProps,
+  onSelect,
+  isSelected,
+}: NodeProps) => (
+  <Box
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    {...getNodeProps()}
+    className={`${buildTreeItemClass(element.id as string)}`}
+    display="flex"
+    sx={{ alignItems: 'center' }}
+  >
+    {isBranch &&
+      (isExpanded ? (
+        <IconButton sx={{ padding: 0 }}>
+          <ExpandMoreIcon />
+        </IconButton>
+      ) : (
+        <IconButton sx={{ padding: 0 }}>
+          <ChevronRightIcon />
+        </IconButton>
+      ))}
+    <Typography
+      component="button"
+      sx={{
+        display: 'flex',
+        alignItems: 'center',
+        gap: '4px',
+        background: isSelected ? deepPurple[50] : 'none',
+        border: 'none',
+        cursor: 'pointer',
+      }}
+      onClick={() => onSelect(element?.id as UUID)}
+    >
+      <FolderIcon fontSize="small" />
+      {element.name}
+    </Typography>
+  </Box>
+);
+
+const TreeView = ({
+  id,
+  header,
+  items,
+  onTreeItemSelect,
+  isLoading = false,
+  onlyShowContainerItems = true,
+}: Props): JSX.Element => {
+  const [showAll, setShowAll] = useState(false);
+  const { mutate: triggerAction } = mutations.usePostItemAction();
+  const itemsToShow = items?.filter((item) =>
+    onlyShowContainerItems ? GRAASP_MENU_ITEMS.includes(item.type) : true,
+  );
+
+  if (isLoading) {
+    return <Skeleton variant="text" />;
+  }
+
+  // types based on TreeView types
+  const onSelect = (value: UUID) => {
+    // trigger player Action for item view
+    triggerAction({ itemId: value, payload: { type: Triggers.ItemView } });
+
+    onTreeItemSelect?.(value);
+  };
+
+  const shownItems = itemsToShow?.slice(
+    0,
+    showAll ? itemsToShow?.length : MAX_NUM_ITEMS,
+  );
+
+  const res = Object.values(getNodeTree(itemsToShow || []))?.[0];
+
+  return (
+    <Box id={id}>
+      {header && (
+        <Typography sx={{ ml: 2 }} variant="body1">
+          {header}
+        </Typography>
+      )}
+      <AccessibleTreeView
+        data={flattenTree({
+          // here there's should be a root item for all children which basically gonna be an empty name
+          name: '',
+          children: res ? [res] : [],
```

Comment:
Can `res` really be `undefined`?

Reply:
Yes, In case we're navigating to a non folder item, SO tree will be {} and so in res will be undefined